client/dashboard/data/site-logs.ts

```diff
@@ -0,0 +1,142 @@
+import wpcom from 'calypso/lib/wp';
+
+export interface SiteLogsAPIResponse {
+	message: string;
+	data: {
+		total_results: number | { value: number; relation: string };
+		logs: ( PHPLogFromEndpoint | ServerLogFromEndpoint )[];
+	};
+}
+
+export interface PHPLogFromEndpoint {
+	timestamp: string;
+	severity: 'User' | 'Warning' | 'Deprecated' | 'Fatal error';
+	message: string;
+	kind: string;
+	name: string;
+	file: string;
+	line: number;
+	atomic_site_id: number;
+}
+
+export interface PHPLog extends Omit< PHPLogFromEndpoint, 'atomic_site_id' > {
+	id: string;
+}
+
+export const LogType = {
+	PHP: 'php',
+	SERVER: 'server',
+} as const;
+
+export type LogType = ( typeof LogType )[ keyof typeof LogType ];
+
+export interface FilterType {
+	[ key: string ]: Array< string >;
+}
+
+export interface ServerLogFromEndpoint {
+	date: string;
+	request_type: 'GET' | 'HEAD' | 'POST' | 'PUT' | 'DELETE';
+	status: '200' | '301' | '302' | '400' | '401' | '403' | '404' | '429' | '500';
+	request_url: string;
+	body_bytes_sent: number;
+	cached: string;
+	http_host: string;
+	http_referer: string;
+	http2: string;
+	http_user_agent: string;
+	http_version: string;
+	http_x_forwarded_for: string;
+	renderer: string;
+	request_completion: string;
+	request_time: string;
+	scheme: string;
+	timestamp: number;
+	type: string;
+	user_ip: string;
+}
+
+export interface ServerLog extends ServerLogFromEndpoint {
+	id: string;
+}
+
+export interface SiteLogsParams {
+	siteId: number | null | undefined;
```

Comment:
I think, `siteId` shouldn't be a field of `SiteLogsParams`. Because then it would be unnecessarily duplicated in `siteLogsQuery()` and its key array.

I'd suggest:

- delete it from this type
- modify the data layer signature to simply 

```ts
function fetchSiteLogs(siteId: number, { /* rest of the params */) {
```

what do you think?

Reply:
I see, yes. The way it's set up, I'm passing in a site id in the query call within the component within the params as well as independently, so I'd agree that this change makes sense - added.